### PR TITLE
[UIE-120] Add append to core-utils.

### PR DESCRIPTION
**Pull request by evrii**
### What
- Created an append function (for lists) in core-utils.
- This should eventually replace the append function in libs/utils.
- In an effort to keep the scope of this PR small, I have not replaced usage of any append functions.

### Why
- libs/Notifications uses append, and I would like to shift that into the components package.
- Lodash/fp is no longer supported, so it did not seem like a good idea to bring append into core-utils

### Testing strategy
- [ ] Used function to add a value to an empty list and to add a value to an existing lists. Tests can be found here:
packages/core-utils/src/list-utils.test.ts

### CODE REVIEW

File: packages/core-utils/src/list-utils.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+export const append = <Type>(value: Type): ((list: Type[]) => Type[]) => {
```
Comment (evrii) [inline packages/core-utils/src/list-utils.ts:13]: A generic, curried version of append that does not rely on lodash/fp.

**Review by cahrens — APPROVED** [review]

File: packages/core-utils/src/list-utils.ts (hunk 1 above)
Comment (msilva-broad) [inline packages/core-utils/src/list-utils.ts:13]: lets add a jsdoc method comment to document intended usage with:
- note that it is "curried" and returns a function expecting array to append item to
- note that it honors immutable pattern (new array returned versus modifying input array)
- example call usage.

File: packages/core-utils/src/list-utils.test.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    expect(result).toEqual([1, 2, 3]);
+  });
+
+  it('adds item to an empty list', () => {
+    // Act
+    const result = append('a')([]);
+
+    // Assert
+    expect(result).toEqual(['a']);
+  });
```
Comment (msilva-broad) [inline packages/core-utils/src/list-utils.test.ts:18]: an additional test to assert that immutable pattern is held up would be nice... simple equality (reference value) of result and orignalValue should be false.

**Review by msilva-broad — APPROVED** [review]
nice work :)

**Pull request merged**